src/App.tsx

```diff
@@ -1,34 +1,139 @@
-/* eslint-disable max-len */
-import React from 'react';
-import 'bulma/css/bulma.css';
-import '@fortawesome/fontawesome-free/css/all.css';
-
-import { TodoList } from './components/TodoList';
+import React, { useState, useEffect } from 'react';
+import { getTodos, getUser } from './api';
+import { TodosList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+
+const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [modalLoading, setModalLoading] = useState(false);
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [userDetails, setUserDetails] = useState<User | null>(null);
+  const [filter, setFilter] = useState('all');
+  const [query, setQuery] = useState('');
+
+  useEffect(() => {
+    const loadTodos = async () => {
+      setLoading(true);
+      const data = await getTodos();
+
+      setTodos(data);
+      setFilteredTodos(data);
+      setLoading(false);
+    };
+
+    loadTodos();
+  }, []);
+
+  const handleShowTodo = async (todo: Todo | null) => {
+    if (!todo) {
+      setSelectedTodo(null);
+      setUserDetails(null);
+
+      return;
+    }
+
+    setSelectedTodo(todo);
+    setModalLoading(true);
+    try {
+      const user = await getUser(todo.userId);
+
+      setUserDetails(user);
+    } catch (error) {
+      return error;
```

Comment:
Returning the error directly from the catch block is not a common practice. Instead, consider logging the error or displaying a user-friendly message to the user.